src/Compilers/Core/Portable/Serialization/ObjectWriter.cs

```diff
@@ -361,10 +361,14 @@ public void Dispose()
             public bool TryGetReferenceId(object value, out int referenceId)
                 => _valueToIdMap.TryGetValue(value, out referenceId);
 
-            public void Add(object value)
+            public void Add(object value, bool isReusable = true)
```

Comment:
might be better to not have the default, and force callers to think about what the right value is to pass.